reference/src/layout/scalars.md

```diff
@@ -37,15 +37,15 @@ The `isize` and `usize` types are pointer-sized signed and unsigned integers.
 They have the same layout as the [pointer types] for which the pointee is
 `Sized`, and are layout compatible with C's `uintptr_t` and `intptr_t` types.
 
+> **Note**: As a result, Rust's `usize` is **not** necessarily layout
+> compatible with C's `size_t`, since `uintptr_t` may be different from
+> `size_t`.
+
 > **Note**: C99 [7.18.2.4](https://port70.net/~nsz/c/c99/n1256.html#7.18.2.4)
 > requires `uintptr_t` and `intptr_t` to be at least 16-bit wide. All 
 > platforms we currently support have a C platform, and as a consequence,
 > `isize`/`usize` are at least 16-bit wide for all of them.
 
-> **Note**: Rust's `usize` and C's `unsigned` types are **not** equivalent. C's
-> `unsigned` is at least as large as a short, allowed to have padding bits, etc.
```

Comment:
No strong opinion from my side. I am not sure removing this information will help anyone though. This is a rather long and somewhat messy list of notes anywhere, maybe it can be organized better?